hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -826,6 +828,28 @@ private VolumeList getVolumesByUser(String userNameKey)
     return count;
   }
 
+  @Override
+  public List<String> getMultipartUploadKeys(
+      String volumeName, String bucketName, String prefix) throws IOException {
+    List<String> response = new ArrayList<>();
+    TableIterator<String, ? extends KeyValue<String, OmMultipartKeyInfo>>
+        iterator = getMultipartInfoTable().iterator();
+
+    String prefixKey =
+        OmMultipartUpload.getDbKey(volumeName, bucketName, prefix);
+    iterator.seek(prefixKey);
+
+    while (iterator.hasNext()) {
```

Comment:
Here, we need to consider table cache also now HA/Non-HA code path is merged (HDDS-1909)

Reply:
Can you please help me to understand how should it be done? What about if an MPU is finished and deleted? How is it cached? I think I can't return with (cached values + db values) because the deletions....